carbon-design-system/carbon
# feat(floating-ui): add feature flag for dynamic floating-ui styles

Closes #17018 

This PR introduces a new feature flag (`enable-v12-dynamic-floating-styles`) to separate the `@floating-ui` logic. By turning on the flag, components will have dynamically positioned floating styles without the collision detection currently in place. 

To have the collision detect logic, adopter will set the `autoAlign` prop to true which will automatically include the dynamic styling - so the feature flag is not necessary in this case.

#### Changelog

**New**

- `enable-v12-dynamic-floating-styles` feature flag

**Changed**

- add checks in current components utilitizing `@floating-ui` to separate out the style and middleware logic

#### Testing / Reviewing

- go to deploy preview, `Feature flags` section
- go to `Dynamic floating styles` folder and inspect the components and see that the styles are set dynamically
